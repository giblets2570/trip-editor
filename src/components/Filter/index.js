import React, { Component } from 'react'
import { connect } from 'react-redux'
import windowSize from 'react-window-size'
import { DateRangePicker } from 'react-dates'
import {
  Row,
  Col,
  Input,
  Label
} from 'reactstrap'

import './style.css'

import { get } from '../../actions/authActions'
import { fetch, updateFilters } from '../../actions/tripsActions'

class Filter extends Component {
  constructor(){
    super()
    this.state = {}
    this.updateDates = this.updateDates.bind(this)
    this.makeSureUserThere = this.makeSureUserThere.bind(this)
    this.handleChange = this.handleChange.bind(this)
  }
  componentWillMount(){
    this.makeSureUserThere()
  }
  handleChange(event,key) {
    const newFilters = {...this.props.filters, destination: event.target.value}
    this.props.dispatch(updateFilters(newFilters))
  }
  updateDates(startDate, endDate) {
    const newFilters = {...this.props.filters, startDate, endDate}
    this.props.dispatch(updateFilters(newFilters))
  }
  makeSureUserThere() {
    if(this.props.user.role === 'admin'){
      const user_id = this.props.match ? this.props.match.params.id : null;
      if(user_id){
        const pagesUser = this.props.users.find((user) => user._id === user_id)
        if(!pagesUser){
          this.props.dispatch(get(user_id))
        }
      }
    }
  }
  render() {
    let ifAdmin = null

    if(this.props.user.role === 'admin'){
      const user_id = this.props.match ? this.props.match.params.id : null;
      const pagesUser = this.props.users.find((user) => user._id === user_id)
      if(pagesUser){
        ifAdmin = (<h2 className="ifAdminHeader">Showing trips for {pagesUser.name}</h2>)
      }
    }
    return (
      <Row>
        <Col lg="12">
          {ifAdmin}
        </Col>
        <Col lg="4" xs="12">
          <Label 
            for="DestinationFilter">Search for your destination</Label>
          <Input
            type="text"
            value={this.props.filters.destination}
            name="destinationFilter"
            id="DestinationFilter"
            onChange={(e) => this.handleChange(e)}
            placeholder="Paris" />
        </Col>
        <Col lg="4" xs="12">
          <Label 
            for="Range">Only show trips that start between dates</Label>
          <br/>
          <DateRangePicker
            startDate={this.props.filters.startDate} // momentPropTypes.momentObj or null,
            endDate={this.props.filters.endDate} // momentPropTypes.momentObj or null,
            displayFormat="DD/MM/YYYY"
            onDatesChange={({ startDate, endDate }) => this.updateDates(startDate, endDate)} // PropTypes.func.isRequired,
            focusedInput={this.state.focusedInput} // PropTypes.oneOf([START_DATE, END_DATE]) or null,
            onFocusChange={focusedInput => this.setState({ focusedInput })} // PropTypes.func.isRequired,
            withFullScreenPortal={this.props.windowWidth < 768}
            orientation={this.props.windowWidth < 768 ? "vertical" : "horizontal"}
          />
        </Col>
        <Col lg="4" xs="12">
          <Label check>
            <Input 
              type="checkbox" 
              onClick={() => this.setState({hidePast: !this.state.hidePast})}/>{' '}
            Hide trips that are in the past
          </Label>
        </Col>
      </Row>
    )
  }
}

export default connect((store) => {
  return {
    filters: store.trips.filters,
    user: store.auth.user,
    users: store.auth.users
  }
})(windowSize(Filter))