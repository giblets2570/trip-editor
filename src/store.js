import { compose, applyMiddleware, createStore } from "redux"
import { persistStore, autoRehydrate } from "redux-persist"

import { createLogger }  from "redux-logger"
import thunk from "redux-thunk"
import promise from "redux-promise-middleware"

import reducer from "./reducers"

const middleware = applyMiddleware(promise(), thunk, createLogger())

const store = createStore(
	reducer, 
	undefined,
	compose(
		middleware,
		// autoRehydrate()
	)
)

// persistStore(store);

export default store;