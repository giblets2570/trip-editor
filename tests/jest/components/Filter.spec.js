import React from 'react';
import renderer from 'react-test-renderer';
import Filter from '../../../src/components/Filter';
import store from '../../../src/store'
import { Provider } from 'react-redux'

describe('Filter (Snapshot)', () => {
  it('Filter renders', () => {
  	const match = {
  		params: {
  			id: '1234'
  		}
  	}
    let wrapper = shallow(
      <Filter match={match} store={mockStore} />
    );
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.prop('filters').destination).toEqual("");
    expect(wrapper.prop('filters').startDate).toEqual(null);
    expect(wrapper.prop('filters').endDate).toEqual(null);
    expect(wrapper.prop('user')).toEqual({role: null});
    expect(wrapper.prop('users')).toEqual([]);
    expect(wrapper.prop('match').params.id).toEqual('1234')
  });
});