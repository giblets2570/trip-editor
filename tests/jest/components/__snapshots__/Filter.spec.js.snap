// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Filter (Snapshot) Filter renders 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <_class
    dispatch={[Function]}
    filters={
      Object {
        "destination": "",
        "endDate": null,
        "startDate": null,
      }
    }
    match={
      Object {
        "params": Object {
          "id": "1234",
        },
      }
    }
    store={
      Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      }
    }
    storeSubscription={
      Subscription {
        "listeners": Object {
          "notify": [Function],
        },
        "onStateChange": [Function],
        "parentSub": undefined,
        "store": Object {
          "clearActions": [Function],
          "dispatch": [Function],
          "getActions": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        },
        "unsubscribe": null,
      }
    }
    user={
      Object {
        "role": null,
      }
    }
    users={Array []}
  />,
  "nodes": Array [
    <_class
      dispatch={[Function]}
      filters={
        Object {
          "destination": "",
          "endDate": null,
          "startDate": null,
        }
      }
      match={
        Object {
          "params": Object {
            "id": "1234",
          },
        }
      }
      store={
        Object {
          "clearActions": [Function],
          "dispatch": [Function],
          "getActions": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        }
      }
      storeSubscription={
        Subscription {
          "listeners": Object {
            "notify": [Function],
          },
          "onStateChange": [Function],
          "parentSub": undefined,
          "store": Object {
            "clearActions": [Function],
            "dispatch": [Function],
            "getActions": [Function],
            "getState": [Function],
            "replaceReducer": [Function],
            "subscribe": [Function],
          },
          "unsubscribe": null,
        }
      }
      user={
        Object {
          "role": null,
        }
      }
      users={Array []}
    />,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <Connect(_class)
        match={
          Object {
            "params": Object {
              "id": "1234",
            },
          }
        }
        store={
          Object {
            "clearActions": [Function],
            "dispatch": [Function],
            "getActions": [Function],
            "getState": [Function],
            "replaceReducer": [Function],
            "subscribe": [Function],
          }
        }
      />,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": Connect {
        "_reactInternalInstance": [Circular],
        "context": Object {
          "store": undefined,
          "storeSubscription": undefined,
        },
        "notifyNestedSubs": [Function],
        "props": Object {
          "match": Object {
            "params": Object {
              "id": "1234",
            },
          },
          "store": Object {
            "clearActions": [Function],
            "dispatch": [Function],
            "getActions": [Function],
            "getState": [Function],
            "replaceReducer": [Function],
            "subscribe": [Function],
          },
        },
        "propsMode": true,
        "refs": Object {},
        "renderCount": 0,
        "selector": Object {
          "error": null,
          "props": Object {
            "dispatch": [Function],
            "filters": Object {
              "destination": "",
              "endDate": null,
              "startDate": null,
            },
            "match": Object {
              "params": Object {
                "id": "1234",
              },
            },
            "store": Object {
              "clearActions": [Function],
              "dispatch": [Function],
              "getActions": [Function],
              "getState": [Function],
              "replaceReducer": [Function],
              "subscribe": [Function],
            },
            "user": Object {
              "role": null,
            },
            "users": Array [],
          },
          "run": [Function],
          "shouldComponentUpdate": false,
        },
        "setWrappedInstance": [Function],
        "state": Object {},
        "store": Object {
          "clearActions": [Function],
          "dispatch": [Function],
          "getActions": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        },
        "subscription": Subscription {
          "listeners": Object {
            "notify": [Function],
          },
          "onStateChange": [Function],
          "parentSub": undefined,
          "store": Object {
            "clearActions": [Function],
            "dispatch": [Function],
            "getActions": [Function],
            "getState": [Function],
            "replaceReducer": [Function],
            "subscribe": [Function],
          },
          "unsubscribe": null,
        },
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
        "version": 0,
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <_class
          dispatch={[Function]}
          filters={
            Object {
              "destination": "",
              "endDate": null,
              "startDate": null,
            }
          }
          match={
            Object {
              "params": Object {
                "id": "1234",
              },
            }
          }
          store={
            Object {
              "clearActions": [Function],
              "dispatch": [Function],
              "getActions": [Function],
              "getState": [Function],
              "replaceReducer": [Function],
              "subscribe": [Function],
            }
          }
          storeSubscription={
            Subscription {
              "listeners": Object {
                "notify": [Function],
              },
              "onStateChange": [Function],
              "parentSub": undefined,
              "store": Object {
                "clearActions": [Function],
                "dispatch": [Function],
                "getActions": [Function],
                "getState": [Function],
                "replaceReducer": [Function],
                "subscribe": [Function],
              },
              "unsubscribe": null,
            }
          }
          user={
            Object {
              "role": null,
            }
          }
          users={Array []}
        />,
        "_debugID": 2,
        "_renderedOutput": <_class
          dispatch={[Function]}
          filters={
            Object {
              "destination": "",
              "endDate": null,
              "startDate": null,
            }
          }
          match={
            Object {
              "params": Object {
                "id": "1234",
              },
            }
          }
          store={
            Object {
              "clearActions": [Function],
              "dispatch": [Function],
              "getActions": [Function],
              "getState": [Function],
              "replaceReducer": [Function],
              "subscribe": [Function],
            }
          }
          storeSubscription={
            Subscription {
              "listeners": Object {
                "notify": [Function],
              },
              "onStateChange": [Function],
              "parentSub": undefined,
              "store": Object {
                "clearActions": [Function],
                "dispatch": [Function],
                "getActions": [Function],
                "getState": [Function],
                "replaceReducer": [Function],
                "subscribe": [Function],
              },
              "unsubscribe": null,
            }
          }
          user={
            Object {
              "role": null,
            }
          }
          users={Array []}
        />,
      },
      "_renderedNodeType": 1,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <Connect(_class)
    match={
      Object {
        "params": Object {
          "id": "1234",
        },
      }
    }
    store={
      Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      }
    }
  />,
}
`;
